pkg/auth/manager.go

```diff
@@ -73,122 +75,96 @@ func newAuthManager(authHandlers []authHandler, dpAuthenticator datapathAuthenti
 	}
 
 	return &authManager{
+		signalChannel:         signalChannel,
 		authHandlers:          ahs,
 		datapathAuthenticator: dpAuthenticator,
 		ipCache:               ipCache,
+		pending:               make(map[AuthKey]struct{}),
 	}, nil
 }
 
-func (a *authManager) AuthRequired(dn *monitor.DropNotify, ci *monitor.ConnectionInfo) {
-	if err := a.authRequired(dn, ci); err != nil {
-		log.WithError(err).Warning("auth: Failed to authenticate request")
-		return
+// start receives auth required signals from the signal channel and spawns
+// a new go routine for each authentication request
+func (a *authManager) start() {
+	go func() {
+		for key := range a.signalChannel {
+			if a.markPendingAuth(key) {
+				go func(key AuthKey) {
+					defer a.clearPendingAuth(key)
+
+					// Check if the auth is actually required, as we might have
+					// updated the authmap since the datapath issued the auth
+					// required signal.
+					if a.datapathAuthenticator.checkAuthenticated(key) {
+						log.Debugf("auth: Already authenticated, skipped authentication for key %v", key)
```

Comment:
should we be returning here?

Reply:
Yes, messed this up inlining the former `Authenticate()` function here! Thanks for noticing :-)